File: Backend/app/api/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel

from app.api.schemas.chat import ChatSourceResponse


class ConversationCreate(BaseModel):
    title: str | None = None


class ConversationResponse(BaseModel):
    id: str
    title: str


class MessageResponse(BaseModel):
    id: str
    conversation_id: str
    role: str
    content: str
    sources: list[ChatSourceResponse]
    created_at: datetime

File: Frontend/types/conversation.ts
import type { ChatSource } from "./source";

export interface Conversation {
  id: string;
  title: string;
  active_document_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConversationMessage {
  id: string;
  conversation_id: string;
  role: "user" | "assistant";
  content: string;
  sources: ChatSource[];
  document_ids: string[];
  created_at: string;
}
